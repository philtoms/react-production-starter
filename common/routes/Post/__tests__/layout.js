import React from 'react'
import renderer from 'react-test-renderer'

import Layout from '../components/Layout'

describe('Post layout', () => {

  it('should render a loading page', () => {
    const component = renderer.create(
      <Layout title="t" content="c" isLoading/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render a loaded page', () => {
    const component = renderer.create(
      <Layout title="t" content="c"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render an error page', () => {
    const component = renderer.create(
      <Layout error/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

})
