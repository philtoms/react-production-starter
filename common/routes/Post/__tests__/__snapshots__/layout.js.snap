exports[`Post layout should render a loaded page 1`] = `
<div>
  <div>
    <h2
      className="title_1upzcph">
      t
    </h2>
    <p
      className="content_7tduvm">
      c
    </p>
  </div>
</div>
`;

exports[`Post layout should render a loading page 1`] = `
<div>
  <div>
    <h2
      className="loading_1pyb631">
      Loading....
    </h2>
  </div>
</div>
`;

exports[`Post layout should render an error page 1`] = `
<div>
  <h1>
    Page Not Found!
  </h1>
</div>
`;
